### PR TITLE
Add the copy request button in the networks tab

### PR DESCRIPTION
**Pull request by netos23**
Added the ability to copy requests in the networks tab. In our company, we faced the problem that after the release of the new version of devtools, it became inconvenient to copy requests. In most cases, it is necessary to copy the request and not the response. I added this feature on the principle of copying the answers and also wrote a test.

- added the ability to copy requests in the networks tab
- fixed json request for network_request_inspector_test.dart
before
after

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by kenzieschmoll** [comment]
Thanks for the PR - can you add a screenshot or a video of the feature?

**Comment by netos23** [comment]
> Thanks for the PR - can you add a screenshot or a video of the feature?

Thanks for your reply, added screenshots before and after

File: packages/devtools_app/lib/src/screens/network/network_request_inspector.dart
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/network/network_request_inspector.dart:60]: Is this `Row` necessary here?

File: packages/devtools_app/test/test_data/network.dart
Comment (bkonyi) [inline packages/devtools_app/test/test_data/network.dart:252]: Why did this data need to be changed?

**Review by bkonyi — COMMENTED** [review]
Thanks for the contribution! For completeness, we might want to add this for the response as well.

Just a couple of comments.

File: packages/devtools_app/test/test_data/network.dart
Comment (netos23) [reply packages/devtools_app/test/test_data/network.dart:252]: The problem is that the encoded request body contains single quotes that cannot be decoded through the jsonDecode method. I replaced the single quotes with double ones
Before:
`    {
      title: 'foo',
      body: 'bar',
      userId: 1,
    }
    `
After:
`{
 "title": "foo", "body": "bar", "userId": 1
}
 `

File: packages/devtools_app/lib/src/screens/network/network_request_inspector.dart
Comment (netos23) [reply packages/devtools_app/lib/src/screens/network/network_request_inspector.dart:60]: Technically not. The `Row` is used for positioning widgets and can be replaced by an `Align`. I wrote this piece of code similarly to the code below.

File: packages/devtools_app/test/test_data/network.dart
Comment (bkonyi) [reply packages/devtools_app/test/test_data/network.dart:252]: I think this change is breaking some other tests.

File: packages/devtools_app/lib/src/screens/network/network_request_inspector.dart
Comment (bkonyi) [reply packages/devtools_app/lib/src/screens/network/network_request_inspector.dart:60]: I'd replace this with an `Align` then (we should probably do that below as well).

File: packages/devtools_app/test/test_data/network.dart
Comment (netos23) [reply packages/devtools_app/test/test_data/network.dart:252]: I have studied this issue.  Single quotes for names and string literals are supported only in JSON 5 and dart convert's json decode does not support JSON 5, therefore the request body with single quotes is invalid. Unfortunately, this change really breaks some tests, specifically curl_command_test.dart, I suggest making json valid for this test and modifying it

File: packages/devtools_app/lib/src/screens/network/network_request_inspector.dart
Comment (netos23) [reply packages/devtools_app/lib/src/screens/network/network_request_inspector.dart]: Thank you for your review. I have fixed this error

File: packages/devtools_app/test/test_data/network.dart
Comment (bkonyi) [reply packages/devtools_app/test/test_data/network.dart:252]: Yeah, that seems like a reasonable change to make as long as the tests still pass :-)

File: packages/devtools_app/test/network/network_request_inspector_test.dart
Comment (bkonyi) [inline packages/devtools_app/test/network/network_request_inspector_test.dart:121]: FYI, `dart format` might complain about this if you haven't already formatted all your changes.

**Review by bkonyi — APPROVED** [review]
Looks good to me once tests are passing. Thanks for the contribution!

**Comment by bkonyi** [comment]
@kenzieschmoll can you also review? I think we need approval from two Googlers so g3 won't be upset when we try and roll this change in.

**Review by kenzieschmoll — APPROVED** [review]
Also LGTM once tests are passing - thanks for the contribution!

File: packages/devtools_app/test/network/network_request_inspector_test.dart
Comment (netos23) [reply packages/devtools_app/test/network/network_request_inspector_test.dart]: Sorry I didn't pay attention. I have reformatted the files.

**Pull request merged**
